FinniversKit/Sources/Components/BrazePromotionView/BrazePromotionView.swift

```diff
@@ -0,0 +1,249 @@
+            closeButton.backgroundColor = UIColor.clear
+            return
+        }
+        NSLayoutConstraint.deactivate(stackViewConstraintsNoImage)
+        NSLayoutConstraint.activate(stackViewConstraintsImage)
+        backgroundView.addSubview(remoteImageView)
+        remoteImageView.dataSource = imageDatasource
+        NSLayoutConstraint.activate([
+            remoteImageView.widthAnchor.constraint(equalTo: backgroundView.widthAnchor, multiplier: 0.3),
+            remoteImageView.topAnchor.constraint(equalTo: backgroundView.topAnchor),
+            remoteImageView.trailingAnchor.constraint(equalTo: backgroundView.trailingAnchor),
+            remoteImageView.bottomAnchor.constraint(equalTo: backgroundView.bottomAnchor)
+        ])
+        remoteImageView.loadImage(
+            for: imageUrl,
+            imageWidth: 100,
+            fallbackImage: UIImage(named: .noImage)
+        )
+    }
+}
+
+// MARK: - Private functions
+extension BrazePromotionView {
+    private func setup() {
+        translatesAutoresizingMaskIntoConstraints = false
+
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(viewWasTapped))
+        addGestureRecognizer(tapGesture)
+
+        addSubview(largeShadowView)
+        largeShadowView.fillInSuperview()
+        largeShadowView.addSubview(smallShadowView)
+        smallShadowView.fillInSuperview()
+        smallShadowView.addSubview(backgroundView)
+        backgroundView.fillInSuperview()
+        backgroundView.addSubview(verticalStackView)
+        backgroundView.addSubview(closeButton)
+        verticalStackView.addArrangedSubviews([titleLabel, textLabel, primaryButton])
+
+        NSLayoutConstraint.activate([
+            closeButton.topAnchor.constraint(equalTo: backgroundView.topAnchor, constant: .spacingS),
+            closeButton.trailingAnchor.constraint(equalTo: backgroundView.trailingAnchor, constant: -.spacingS),
+            closeButton.widthAnchor.constraint(equalToConstant: buttonSize),
+            closeButton.heightAnchor.constraint(equalToConstant: buttonSize)
+        ])
+    }
+
+    @objc private func primaryButtonTapped() {
+        delegate?.brazePromotionView(self, didSelect: .primary)
+    }
+
+    @objc private func viewWasTapped() {
+        delegate?.brazePromotionViewTapped(self)
+    }
+
+    @objc private func handleTapOnCloseButton() {
+        delegate?.brazePromotionView(self, didSelect: .secondary)
+    }
+}
+
+// MARK: - Private extensions
+
+private extension UIColor {
+    static var shadowColor: UIColor {
+        return UIColor(hex: "475569")
+    }
+
+    static var bgColor: UIColor {
+        return .dynamicColor(defaultColor: .milk, darkModeColor: .blueGray700)
+    }
+}
+
+private extension Button {
+    func configure(withTitle title: String?) {
+        if let title = title {
+            setTitle(title, for: .normal)
+        } else {
+            isHidden = true
+        }
+    }
```

Comment:
This code won't present the button again, if the view is supposed to be reconfigurable. It should rather be like this:

```swift
func configure(withTitle title: String?) {
    setTitle(title, for: .normal)
    isHidden = title == nil
}
```